### PR TITLE
Class07

### PR DESCRIPTION
**Pull request by RivaD2**
- [x] Used modular system with express Router object
- [x] Successfully connected to server
- [x] Successfully connected to MongoDB
- [x] Sucessfully tested routes via Compass
- [x] Added the following middleware files to `middleware.js`.  Imported all middleware into my `server.js` and set them up to run at the application level for all routes.
     - `timestamp.js`: this puts the current timestamp (formatted like a proper date) on the request object in a property called 
      requestTime.
    - `logger.js`:Executes a `console.log()` containing the request path, method, and the requestTime property of the request 
      object
    - `404.js`: Sends a 404/Not-Found message as the response (does not call `.next()`)
    - `500.js`: Sends a 500/Server Error message as the response (does not call .`next()`)
- [x] Created `routes` folder  for modularity and added a `categories.js` and a `products.js` where all routes are defined
- [x] Both route files(for products and categories) include the following CRUD routes:
    - router.post('/products', (req, res) => {}) … uses the object in the body of the request to create a new “record”
    - router.get('/products', (req, res) => {})
    - router.get('/products/:id', (req, res) => {})
    - router.put('/products/:id', (req, res) => {}) … uses the object in the body to replace the record with the :id specified
    - router.delete('/products/:id', (req, res) => {}) … deletes the record with the :id specified
- [x] feature tasks completed 
- [x] updated top level README.md
**- [] testing** IN PROGRESS

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +29,20 @@
...
+- Start server:
+```
+    /* give it a port number and optionally pass a function to call when app
+     starts listening on given port*/
+
+const port = process.env.PORT || 3000;
+app.listen(port, () => console.log(`Listening on port ${port}`));
+
+```
+
```
Comment (adrianhuebner) [inline README.md:44]: I would like to see a note here about `node index.js` or whatever it needs to be to run this application

File: README.md [hunk 2]
```diff
@@ -25,6 +29,20 @@
...
+
```
Comment (adrianhuebner) [inline README.md:47]: I would also add another section with the title `###env requirments` where you put everything that needs to be added like PORT, etc

File: lib/server.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
+'use strict'
+require('dotenv').config();;
+const express = require('express');
+const app = express()
```
Comment (adrianhuebner) [inline lib/server.js:4]: This is why router doesn't work, if you switch this to router it should work

Comment (RivaD2) [reply lib/server.js:4]: Roger that! Fixed it overall! Still not seeing my `logger` middleware or other middleware in console...hmm. That works before when using app.use. I shall investigate! 👍

**Pull request merged**